src/cmd/initContainer.go

```diff
@@ -400,7 +400,41 @@ func mountBind(containerPath, source, flags string) error {
 func redirectPath(containerPath, target string, folder bool) error {
 	logrus.Debugf("Redirecting %s to %s", containerPath, target)
 
-	err := os.Remove(containerPath)
+	targetInfo, err := os.Lstat(target)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("target %s was not found in the container", target)
+		}
+		return fmt.Errorf("failed to lstat %s", target)
+	}
+	isSymlink := (targetInfo.Mode() & os.ModeSymlink) != 0
+
+	// If the target is a link, follow it. If needed, repeat the process.
+	for isSymlink {
+		logrus.Debugf("Path %s is a symlink. Resolving.", target)
+		newTarget, err := os.Readlink(target)
+		if err != nil {
+			return fmt.Errorf("Could not read link %s", target)
+		}
+
+		// Try to find the file inside of the container
+		target = fmt.Sprintf("/run/host%s", newTarget)
```

Comment:
Yes, I am aware of this shortcoming. I don't deny that I put this patch together rather quickly and it is not *the perfect* solution. I currently don't have a solution for this ready but I might at least add to the comment a `TODO` to highlight the need to make this mechaninsm more flexible.

Reply:
One solution might be to check if the symlink is absolute with `filepath.IsAbs()`. If not, join it to the directory it resides in (`filepath.Dir()`). Then join the resulting path to `"/run/host"`. You should join paths with `filepath.Join()` instead of `fmt.Sprintf()`.

Reply:
I incorporated your suggestion in the latest iteration of the PR. What do you think?